orca-queue/src/main/kotlin/com/netflix/spinnaker/orca/q/handler/CompleteStageHandler.kt

```diff
@@ -165,7 +165,14 @@ class CompleteStageHandler(
 
     val graph = StageGraphBuilder.afterStages(this)
     builder().onFailureStages(this, graph)
+
     val onFailureStages = graph.build().toList()
+    onFailureStages.forEachIndexed { index, stage ->
+      if (index > 0) {
+        // all on failure stages should be run linearly
+        graph.connect(onFailureStages.get(index - 1), stage)
```

Comment:
Without this, we run the risk of one of the stages completing and notifying it's parent ... and completing the execution with another stage still running in parallel.

My expectation is that > 1 onFailure stages will run linearly (they did previously).

Reply:
I'd like to address the root issue as well (not in this PR). The execution should not complete if anything is still in a `RUNNING` state and if it does that's a bug.